### PR TITLE
[MRG] Fix pep8 compliance

### PR DESCRIPTION
**Pull request by betatim**
Clean up some pep8 warnings and additional tests.

Added an additional test and logic in `check_dimension` to get coverage
for raising the exception.

### CODE REVIEW

**Comment by glouppe** [comment]
:+1:

File: skopt/learning/gaussian_process/kernels.py [hunk 1]
```diff
@@ -287,12 +287,13 @@ def gradient_x(self, x, X_train):
...
         )
 
+
 class Product(Kernel, sk_Product):
 
     def gradient_x(self, x, X_train):
         x = np.asarray(x)
+        x = np.expand_dims(x, axis=0)
         X_train = np.asarray(X_train)
-        X = np.expand_dims(x, axis=0)
```
Comment (betatim) [inline skopt/learning/gaussian_process/kernels.py]: Was this a typo? Capital X instead of x? I assumed that it was and fixed it accordingly.

Comment (glouppe) [inline skopt/learning/gaussian_process/kernels.py]: indeed, seems like a good catch

Comment (betatim) [inline skopt/learning/gaussian_process/kernels.py]: is there a test we can add that would have detected this? To stop the :beetle: :ghost: from coming back

**Comment by glouppe** [comment]
Merging this, nothing controversial to be discussed.

**Pull request merged**